cohortextractor/study_definition.py

```diff
@@ -136,14 +126,24 @@ def assert_backend_is_configured(self):
                 "Cannot extract data as no DATABASE_URL environment variable defined"
             )
 
-    def validate_study_definition(self, covariate_definitions):
+    def validate_study_definition(self):
         # As a crude way of error checking we construct a TPP backend with a
         # dummy database URL. We immediately discard the backend instance, but
         # the process of constructing it should trigger any problems with the
         # study definition.
-        database_url = "mssql://localhost/dummy"
-        Backend = self.get_backend_for_database_url(database_url)
-        Backend(database_url, covariate_definitions)
+        self.create_backend("mssql://localhost/dummy")
+
+    def create_backend(self, database_url=None):
```

Comment:
Doc string - I am slightly unclear what this is going? Is this the pointer for backend?

Reply:
Sorry, Caroline, I don't follow your question.

I've introduced this method to remove a bit of duplication where we look up the backend type based on URL and then instantiate it.

Reply:
sorry - i was just asking for a docstring as I wasn't sure what this was doing on first glance. Thanks for clarifying. It makes sense.

Reply:
Thanks, understood. I've added an explanatory comment.